cfme/v2v/migration_throttling.py

```diff
@@ -0,0 +1,61 @@
+import attr
+from navmazing import NavigateToAttribute
+from widgetastic.utils import WaitFillViewStrategy
+from widgetastic_patternfly import Button
+from widgetastic_patternfly import Input
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+
+
+class MigrationSettingsView(BaseLoggedInPage):
+    fill_strategy = WaitFillViewStrategy("20s")
+    migration_per_conv_host = Input(id='max_concurrent_tasks_per_host')
+    migration_per_provider = Input(id='max_concurrent_tasks_per_ems')
+    apply_btn = Button("Apply")
+
+    @property
+    def is_displayed(self):
+        nav_menu = ["Compute", "Migration", "Migration Settings"]
+        return self.logged_in_as_current_user and self.navigation.currently_selected == nav_menu
+
+
+@attr.s
+class MigrationThrottling(BaseEntity):
+    """Class representing Migration throttling
+        Args:
+            migration_per_conv_host: maximum concurrent migration per conversion host
+            migration_per_provider: maximum concurrent migrtaion per provider
+    """
+    migration_per_conv_host = attr.ib(default=None)
+    migration_per_provider = attr.ib(default=None)
+
+
+@attr.s
+class MigrationThrottlingCollection(BaseCollection):
```

Comment:
Is there any `create` method for such a collection? It looks like `set_max_migration_per_conv_host` is being used like a `create` method.

Reply:
![Screenshot from 2019-07-26 12-23-58](https://user-images.githubusercontent.com/3866976/61976291-4512d100-afa0-11e9-96c5-61db799aa136.png)

Setting max values is the only operation permitted from this UI .

Reply:
John is getting at an important issue here, but sort of for the wrong reasons.

A BaseCollection/BaseEntity derived class, modeling a collection of `things` :tm: in MIQ does not necessarily need a `create` method.  I would agree with him, that if this were a proper collection, that this method should be `create`.

But the real issue is that this isn't a collection, or an entity, its simply a configuration setting for Migration. Since there is no 'Migration' `BaseEntity` derived class encompassing this part of the appliance, its a setting that is global to the appliance, effectively.

In this case, it should inherit from `NavigatableMixin`, and not `BaseEntity`, like the appliance level settings:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/configure/settings.py

Additionally, this view already exists, for Conversion Hosts:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/v2v/conversion_hosts.py#L28

We should not be duplicating the view for this navigation destination or class. There should be a single navigation destination to `'MigrationSettings'`, where its returned `View` class has both tabs. Methods can be placed in the `MigrationSettings` class that inherits `NavigatableMixin`, and to which the navigation destination is registered, to update these settings by filling the tab.

Reply:
Hi @mshriver : Implemented Migration throttling inheriting """NavigatableMixin""".
Please take a look